File: JAVA/java/src/CountSubarraysWithFixedBounds.java
public class CountSubarraysWithFixedBounds {
    public long countSubarrays(int[] A, int minK, int maxK) {
        long res = 0, jbad = -1, jmin = -1, jmax = -1, n = A.length;
        for (int i = 0; i < n; ++i) {
            if (A[i] < minK || A[i] > maxK) jbad = i;
            if (A[i] == minK) jmin = i;
            if (A[i] == maxK) jmax = i;
            res += Math.max(0L, Math.min(jmin, jmax) - jbad);
        }
        return res;
    }

    public static void main(String[] args) {
        CountSubarraysWithFixedBounds ob =new CountSubarraysWithFixedBounds();
        System.out.println(ob.countSubarrays(new int[] {1,3,5,2,7,5},1,5));
    }
}

File: python/Count Subarrays With Fixed Bounds.py
def countSubarrays(nums, minK, maxK):
    res = 0
    minFound = False
    maxFound = False
    start = 0
    minStart = 0
    maxStart = 0
    for i in range(len(nums)):
        num = nums[i]
        if num < minK or num > maxK:
            minFound = False
            maxFound = False
            start = i + 1
        if num == minK:
            minFound = True
            minStart = i
        if num == maxK:
            maxFound = True
            maxStart = i
        if minFound and maxFound:
            res += (min(minStart, maxStart) - start + 1)
    return res
print(countSubarrays([1,3,5,2,7,5],1, 5))